src/monitors/ProfileMonitor.v3

```diff
@@ -0,0 +1,196 @@
+// Copyright 2022 Ben L. Titzer. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// A tree-based profiler
+class ProfileTreeNode(parent: ProfileTreeNode, id: int) {
+    var children: HashMap<int, ProfileTreeNode> = HashMap<int, ProfileTreeNode>.new(int.!<int>, int.==);
```

Comment:
We'll need to eventually keep track of functions in other modules, so an `int` id won't be sufficient. Also, the overhead of a `HashMap` might be a lot, and I suspect most nodes have only a few children. So maybe just a vector or list of `WasmFunction`?

Reply:
Done